lib/spack/spack/cmd/install.py

```diff
@@ -223,8 +223,12 @@ def install_spec(cli_args, kwargs, abstract_spec, spec):
         # handle active environment, if any
         env = ev.get_env(cli_args, 'install')
         if env:
-            env.install(abstract_spec, spec, **kwargs)
-            env.write()
+            with env.write_transaction():
+                concrete = env.concretize_and_add(
+                    abstract_spec, spec)
+                env.write(regenerate_views=False)
+            env._install(concrete, **kwargs)
+            env.regenerate_views()
```

Comment:
If several processes try to regenerate views at once is that ok?  This seems inconsistent with the call inside `install()` below, where `regenereate_views()` is inside the write transaction.  Shouldn't these at least both do the same thing?

Reply:
See https://github.com/spack/spack/pull/14621#discussion_r371965920

Reply:
I should have also clarified: I was operating on the assumption that it should be OK for multiple processes to run `regenerate_views` at the same time (though that may be expensive).